pkg/training/trainer_ray.go

```diff
@@ -0,0 +1,373 @@
+	log.Debugf("PriorityClassName: %v", pc)
+
+	return pc
+}
+
+// RayJob Job trainer
+type RayJobTrainer struct {
+	client       *kubernetes.Clientset
+	RayJobClient *versioned.Clientset
+	trainerType  types.TrainingJobType
+	// check if it's enabled
+	enabled bool
+}
+
+// NewRayJobTrainer
+func NewRayJobTrainer() Trainer {
+	// get rayjob operator client call rayjob operator api
+	enable := false
+	RayJobClient := versioned.NewForConfigOrDie(config.GetArenaConfiger().GetRestConfig())
+	_, err := config.GetArenaConfiger().GetAPIExtensionClientSet().ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), k8saccesser.RayJobCRDName, metav1.GetOptions{})
+	if err == nil {
+		log.Debugf("RayJobTrainer is enabled")
+		enable = true
+	} else {
+		log.Debugf("RayJobTrainer is disabled,reason: %v", err)
+	}
+	log.Debugf("Succeed to init RayJobTrainer")
+	return &RayJobTrainer{
+		RayJobClient: RayJobClient,
+		client:       config.GetArenaConfiger().GetClientSet(),
+		trainerType:  types.RayJob,
+		enabled:      enable,
+	}
+}
+
+// IsEnabled is used to get the trainer is enable or not
+func (rjt *RayJobTrainer) IsEnabled() bool {
+	return rjt.enabled
+}
+
+// Get the type
+func (rjt *RayJobTrainer) Type() types.TrainingJobType {
+	return rjt.trainerType
+}
+
+// check if it's ray job
+func (rjt *RayJobTrainer) IsSupported(name, ns string) bool {
+	if !rjt.enabled {
+		return false
+	}
+	isRayJob := false
+	_, err := rjt.GetTrainingJob(name, ns)
+	if err != nil {
+		return isRayJob
+	}
+	return !isRayJob
+}
+
+// Get the training job from cache or directly
+func (rjt *RayJobTrainer) GetTrainingJob(name, namespace string) (TrainingJob, error) {
+	rayJob, err := k8saccesser.GetK8sResourceAccesser().GetRayJob(rjt.RayJobClient, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+	if err := CheckJobIsOwnedByTrainer(rayJob.Labels); err != nil {
+		return nil, err
+	}
+	// 2. Find the pod list, and determine the pod of the ray job
+	rayClusterPods, err := k8saccesser.GetK8sResourceAccesser().ListPods(namespace, fmt.Sprintf("release=%v,app=%v", name, rjt.Type()), "", nil)
+	if err != nil {
+		return nil, err
+	}
+	rayJobPods, err := k8saccesser.GetK8sResourceAccesser().ListPods(namespace, fmt.Sprintf("%v=%v", labelRayJobName, name), "", nil)
+	if err != nil {
+		return nil, err
+	}
+	allPods := append(rayClusterPods, rayJobPods...)
+	pods, chiefPod := getPodsOfRayJob(rjt, rayJob, allPods)
+	// 3. Find the other resources, like statefulset,job
+	return &RayJob{
+		BasicJobInfo: &BasicJobInfo{
+			resources: podResources(pods),
+			name:      name,
+		},
+		rayJob:      rayJob,
+		chiefPod:    chiefPod,
+		pods:        pods,
+		trainerType: rjt.Type(),
+	}, nil
+
+}
+
+func (rjt *RayJobTrainer) isChiefPod(rayJob *rayv1.RayJob, item *v1.Pod) bool {
+	isChiefPod := false
+
+	if val, ok := item.Labels[rayClusterNodeType]; ok && val == "head" {
+		isChiefPod = true
+	}
+
+	return isChiefPod
+}
```

Comment:
From my understanding, the chief pod of the ray job should be the job pod of the RayJob rather than the head pod of RayCluster, right?

The `rayJob` parameter is unused, do we need to check the job name retrieved from the pod labels and the ray job name are equal?